src/ac_power_flow_residual.jl

```diff
@@ -328,3 +348,25 @@ function _update_residual_values!(
     end
     return
 end
+
+function _find_subnetworks_for_reference_buses(
```

Comment:
Here we are basically taking the output of `PNM.find_subnetworks`, which is a dictionary from the first bus in a subnetwork to a set of all the buses in the subnetwork for all subnetworks, and building a similar structure except the keys are reference buses, right? So not every reference bus gets an entry if there are multiple in a given subnetwork?